opennlp-tools/src/test/java/opennlp/tools/tokenize/TokenizerFactoryTest.java

```diff
@@ -163,6 +163,106 @@ void testCustomPatternAndAlphaOpt() throws IOException {
     Assertions.assertTrue(factory.isUseAlphaNumericOptimization());
   }
 
+  void checkCustomPatternForTokenizerME(String lang, String pattern, String sentence,
+      int expectedNumTokens) throws IOException {
+
+    TokenizerModel model = train(new TokenizerFactory(lang, null, true,
+        Pattern.compile(pattern)));
+
+    TokenizerME tokenizer = new TokenizerME(model);
+    String[] tokens = tokenizer.tokenize(sentence);
+
+    Assertions.assertEquals(expectedNumTokens, tokens.length);
+    String[] sentSplit = sentence.replaceAll("\\.", " .").split(" ");
+    for (int i = 0; i < sentSplit.length; i++) {
+      Assertions.assertEquals(sentSplit[i], tokens[i]);
+    }
+  }
+
+  @Test
+  void testCustomPatternForTokenizerMEDeu() throws IOException {
+    String lang = "deu";
+    String pattern = "^[A-Za-z0-9äéöüÄÉÖÜß]+$";
+    String sentence = "Ich wähle den auf S. 183 ff. mitgeteilten Traum von der botanischen Monographie.";
+    checkCustomPatternForTokenizerME(lang, pattern, sentence, 16);
```

Comment:
Side note: 
-
Given the German example with two abbreviated words ("S." and "ff.") one would expect 14 tokens only. However, TokenizerME does not cover abbreviations at all, atm. For this reason, I've opened a new Jira issue and propose an enhancement similar to `SentenceDetectorME` in which abbreviations are correctly used, see OPENNLP-570.

@l-ma Keep it as is for now. We can take it. The new issue [OPENNLP-1525](https://issues.apache.org/jira/browse/OPENNLP-1525) references your improvements with these tests and the aim is to change "16" to "14" in a future PR.

Hooray, you uncovered a missing feature! 🙃

Reply:
Generally speaking: the lack of abbreviation support affects all languages, not only German.

Reply:
Thanks for creating the issue, and +1 to this improvement. Thanks!

Reply:
Thanks for accepting my PR! I'll be happy to help with this new issue or other issues. I should have more time after December 19th.